week-3/Homework/mandatory/5-journey-planner.js

```diff
@@ -7,16 +7,20 @@
   NOTE: only the names should be returned, not the means of transport.
 */
 
-function journeyPlanner() {
+function journeyPlanner(londonLocations, transport) {
+
+
+ return londonLocations.filter( x => x.includes(transport)).map(x => slice(0, 1).join(" "))
```

Comment:
I don't think everything here is quite right.

In the `.map()` function, you're calling a function named `slice` that I don't think exists, so I think this will cause an error. Instead I think you mean `x.slice(0, 1)` instead.

However, I think this is a bit overly complicated. Inside the map function we only need to get the first element from the array. The simplest way to do this would be to use array index syntax. How could we change our code here to do that?

One more tip: *chained* methods like you have here (`.filter(...).map(...)`) are a bit easier to read if you put them on newlines like this:

```js
return londonLocations
  .filter(x => x.includes(transport))
  .map(x => slice(0, 1).join(" "));
```